Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java

```diff
@@ -52,20 +64,257 @@ public KitodoServiceLoader(Class clazz, String modulePath) {
+            DirectoryStream<Path> stream = Files.newDirectoryStream(moduleFolder, "*.jar");
+
+            for (Path f : stream) {
+                JarFile jarFile = new JarFile(f.toString());
+
+                if (hasFrontendFiles(jarFile)) {
+
+                    Enumeration<JarEntry> e = jarFile.entries();
+
+                    URL[] urls = {new URL("jar:file:" + f.toString() + "!/")};
+                    URLClassLoader cl = URLClassLoader.newInstance(urls);
+
+                    while (e.hasMoreElements()) {
+                        JarEntry je = e.nextElement();
+
+                        // TODO: konvention: name der xhtml datei + Form, bspw.: sample.xhtml -> SampleForm.java
+                        // deshalb wird hier auf "Form.class" gesucht
+
+                        if (je.isDirectory() || !je.getName().endsWith("Form.class")) {
+                            continue;
+                        }
+
+                        String className = je.getName().substring(0, je.getName().length() - 6);
+                        className = className.replace('/', '.');
+                        Class c = cl.loadClass(className);
+
+                        String beanName = className.substring(className.lastIndexOf(".") + 1).trim();
+
+                        FacesContext facesContext = FacesContext.getCurrentInstance();
+                        HttpSession session = (HttpSession) facesContext.getExternalContext().getSession(false);
+
+                        session.getServletContext().setAttribute(beanName, c.newInstance());
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            logger.error("Classpath could not be accessed", e.getMessage());
+        }
+    }
+
+    /**
+     * If the found jar files have frontend files, they will be extracted and copied into the frontend folder
+     * of the core module. Before copying, existing frontend files of the same module will be deleted from the
+     * core module. Afterwards the created temporary folder will be deleted as well.
+     */
+    // TODO: Relative Pfade müssen zu absoluten Pfaden angepasst werden
+    private void loadFrontendFilesIntoCore() {
+        Path moduleFolder = FileSystems.getDefault().getPath(modulePath);
+
+        try {
+            DirectoryStream<Path> stream = Files.newDirectoryStream(moduleFolder, "*.jar");
+
+            for (Path f : stream) {
+
+                File loc = new File(f.toString());
+                JarFile jarFile = new JarFile(loc);
+
+                if (hasFrontendFiles(jarFile)) {
+                    String temporaryFolderName = generateRandomString(10);
+                    extractFrontEndFiles(loc.getAbsolutePath(), temporaryFolderName);
+                    String moduleName = extractModuleName(temporaryFolderName);
+                    FileUtils.deleteDirectory(new File("Kitodo/src/main/webapp/pages/" + moduleName));
```

Comment:
Is this value `Kitodo/src/main/webapp/pages` correct? It looks like that this is only working on a develop system.